lib/SILGen/SILGenBridging.cpp

```diff
@@ -1416,7 +1421,11 @@ emitObjCThunkArguments(SILGenFunction &SGF, SILLocation loc, SILDeclRef thunk,
       arg = emitObjCUnconsumedArgument(SGF, loc, arg);
     }
 
-    auto managedArg = SGF.emitManagedRValueWithCleanup(arg);
+    // Don't emit a cleanup if the argument is @in_cxx.
```

Comment:
I tried to use `@in_guaranteed` instead of `@in_cxx` for thunk parameters so that we don't have to make this change, but ran into a few issues.